src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmExcelTemplateExportServiceImpl.java

```diff
@@ -0,0 +1,532 @@
+				this.sheetStylesMap.get(ExcelCellStyle.STYLE_OLIVE_GREEN_WITH_LATERAL_AND_BOTTOM_BORDER));
+			cell.setCellValue(locationDto.getName());
+			count--;
+		}
+		return rowNumIndex;
+	}
+
+	private int writeUnitsSection(final HSSFSheet codesSheet, final int currentRowNum, final List<VariableDetails> units) {
+		int rowNumIndex = currentRowNum;
+		int count = units.size();
+
+		for (final VariableDetails variableDetail : units) {
+			final HSSFRow row = codesSheet.createRow(rowNumIndex++);
+			HSSFCell cell = row.createCell(GermplasmExcelTemplateExportServiceImpl.CODES_SHEET_FIRST_COLUMN_INDEX, CellType.STRING);
+			cell.setCellStyle(count != 1 ? this.sheetStylesMap.get(ExcelCellStyle.STYLE_AQUA_WITH_LATERAL_BORDER) :
+				this.sheetStylesMap.get(ExcelCellStyle.STYLE_AQUA_WITH_LATERAL_AND_BOTTOM_BORDER));
+			cell.setCellValue(variableDetail.getName());
+
+
+			cell = row.createCell(GermplasmExcelTemplateExportServiceImpl.CODES_SHEET_SECOND_COLUMN_INDEX, CellType.STRING);
+			cell.setCellStyle(count != 1 ? this.sheetStylesMap.get(ExcelCellStyle.STYLE_OLIVE_GREEN_WITH_LATERAL_BORDER) :
+				this.sheetStylesMap.get(ExcelCellStyle.STYLE_OLIVE_GREEN_WITH_LATERAL_AND_BOTTOM_BORDER));
+			cell.setCellValue(variableDetail.getDescription());
+			count--;
+		}
+
+		return rowNumIndex;
+	}
+
+	private void writeCodesHeader(final HSSFSheet codesSheet, final int currentRowNum, final String colunmName) {
+		final Locale locale = LocaleContextHolder.getLocale();
+		final HSSFRow row = codesSheet.createRow(currentRowNum);
+
+		HSSFCell cell = row.createCell(GermplasmExcelTemplateExportServiceImpl.CODES_SHEET_FIRST_COLUMN_INDEX, CellType.STRING);
+		cell.setCellStyle(this.sheetStylesMap.get(ExcelCellStyle.HEADING_STYLE_AQUA));
+		cell.setCellValue(this.getMessageSource().getMessage(colunmName, null, locale));
+
+		cell = row.createCell(GermplasmExcelTemplateExportServiceImpl.CODES_SHEET_SECOND_COLUMN_INDEX, CellType.STRING);
+		cell.setCellStyle(this.sheetStylesMap.get(ExcelCellStyle.HEADING_STYLE_OLIVE_GREEN));
+		cell.setCellValue(this.getMessageSource().getMessage("export.germplasm.list.template.description.column", null, locale));
+
+	}
+
+	private HSSFFont buildFont(final String fontName, final int fontHeight, final boolean bold) {
+		final HSSFFont hSSFFont = this.wb.createFont();
+		hSSFFont.setColor(HSSFColor.HSSFColorPredefined.BLACK.getIndex());
+		hSSFFont.setFontName(fontName);
+		hSSFFont.setFontHeightInPoints((short) fontHeight);
+		hSSFFont.setBold(bold);
+		return hSSFFont;
+	}
+
+	private void setCustomColorAtIndex(final IndexedColors indexedColor, final int red, final int green,
+		final int blue) {
+		final HSSFPalette customPalette = this.wb.getCustomPalette();
+		customPalette.setColorAtIndex(indexedColor.index, (byte) red, (byte) green, (byte) blue);
+	}
+
+	public ResourceBundleMessageSource getMessageSource() {
+		return this.messageSource;
+	}
+
+	public enum ExcelCellStyle {
```

Comment:
move this enum before starting the functions

Reply:
Done.